src/test/java/org/elasticsearch/common/logging/log4j/CustomLogfileTest.java

```diff
@@ -0,0 +1,128 @@
+package org.elasticsearch.common.logging.log4j;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import com.carrotsearch.randomizedtesting.annotations.*;
+import com.google.common.io.Files;
+import org.apache.lucene.util.TimeUnits;
+import org.elasticsearch.common.settings.ImmutableSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.env.Environment;
+import org.elasticsearch.env.FailedToResolveConfigException;
+import org.elasticsearch.test.ElasticsearchThreadFilter;
+import org.elasticsearch.test.junit.listeners.LoggingListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import sun.jvm.hotspot.utilities.Assert;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.isEmptyOrNullString;
+import static org.hamcrest.Matchers.isEmptyString;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+/**
+ *
+ * This test class avoids the "java.lang.AssertionError: System properties invariant violated."
+ * which is enforced by Junit Rules defined in AbstractRandomizedTest class.
+ */
+@RunWith(value = com.carrotsearch.randomizedtesting.RandomizedRunner.class)
+public class CustomLogfileTest {
+
+    private RandomizedTest randomizedTest = new RandomizedTest();
```

Comment:
and in case we want to get this change in 1.x, you just need to have a finally block in your test that removes the property that you added rather than doing it from an after method, so you can move to ElasticsearchTestCase too I think.